crates/services/compression/src/service.rs

```diff
@@ -113,8 +127,32 @@ where
         self.sync_notifier
             .send(crate::sync_state::SyncState::NotSynced)
             .ok();
-        // compress the block
-        self.compress_block(block_with_metadata)?;
+
+        if let Some(metrics_manager) = self.metrics_manager {
+            let (compressed_block, compression_duration) = {
+                let start = Instant::now();
+                let compressed_block = self.compress_block(block_with_metadata)?;
+                (compressed_block, start.elapsed().as_millis())
+            };
+
+            metrics_manager.record_compression_duration_ms(compression_duration);
+
+            // we calc the ratio after the compression is done so that these allocations dont affect
+            // the metrics for compression duration
+            if let (Ok(uncompressed), Ok(compressed)) = (
+                postcard::to_allocvec(&block_with_metadata.sealed_block.entity),
+                postcard::to_allocvec(&compressed_block),
+            ) {
+                metrics_manager
+                    .record_compression_ratio(uncompressed.len(), compressed.len());
+            }
```

Comment:
Instead of recording ration here, you can record just compressed size.

The block importer can be responsible for storing the size of uncompressed block. And later on Grafana level we will just calculate ration by ourself.

Note: On the block importer side you also can avoid `postcard` serialization, and get the information about the size from the storage transaction(I suppose it should be possible, but you also need to sum the size of each transaction. If it is hard to do, then we can just serialize it there for now).

Reply:
addressed in 2157152ae3

we can deal with block importer metric in another pr